sycl/include/CL/sycl/detail/pi.h

```diff
@@ -321,23 +405,40 @@ pi_result piQueueFinish(pi_queue command_queue);
 //
 // Memory
 //
-pi_mem piMemCreate( // TODO: change interface to return error code
+pi_mem piMemBufferCreate( // TODO: change interface to return error code
   pi_context   context,
   pi_mem_flags flags,
   size_t       size,
   void *       host_ptr,
   pi_result *  errcode_ret);
 
+pi_mem piMemImageCreate( // TODO: change interface to return error code
+  pi_context              context,
+  pi_mem_flags            flags,
+  const pi_image_format * image_format,
+  const pi_image_desc *   image_desc,
+  void *                  host_ptr,
+  pi_result *             errcode_ret);
+
 pi_result piMemGetInfo(
   pi_mem           mem,
   cl_mem_info      param_name, // TODO: untie from OpenCL
   size_t           param_value_size,
   void *           param_value,
   size_t *         param_value_size_ret);
 
-pi_result piMemRetain(pi_mem);
+pi_result piMemImageGetInfo (
+  pi_mem          image,
+  pi_image_info   param_name,
+  size_t          param_value_size,
+  void *          param_value ,
+  size_t *        param_value_size_ret);
 
-pi_result piMemRelease(pi_mem);
+pi_result piMemRetain(
```

Comment:
Is this (and piMemRelease below) suitable for both buffers and images?

Reply:
That's correct.

Reply:
OK